### PR TITLE
Replace mute/unmute checkmenuitem with imagemenuitem

### PR DESCRIPTION
**Pull request by FreaxMATE**
This replaces the Mute checkmenuitem with a imagemenuitem of mate-volume-status-icon to make it coherent to the mate-volume-applet.

before:
Screenshot at 2021-07-29 14-24-16
after:
Peek 2021-07-29 14-27

It also shows whether you are mutin orunmuting the input or output.
Maybe this (https://github.com/mate-desktop/mate-utils/pull/270#issuecomment-599769512) should really be moved to [mate-desktop](https://github.com/mate-desktop/mate-desktop) as gtk_image_menu_item is used here as well.

### CODE REVIEW

File: mate-volume-control/gvc-stream-status-icon.c [hunk 1]
```diff
@@ -234,11 +230,35 @@ on_status_icon_popup_menu (GtkStatusIcon       *status_icon,
...
 
-        item = gtk_check_menu_item_new_with_mnemonic (_("_Mute"));
-        gtk_check_menu_item_set_active (GTK_CHECK_MENU_ITEM (item),
-                                        mate_mixer_stream_control_get_mute (icon->priv->control));
+        if (mate_mixer_stream_control_get_mute (icon->priv->control))
+        {
+                if (!g_strcmp0(icon->priv->display_name, _("Output")))
+                {
+                        item = gtk_image_menu_item_new_with_mnemonic(_("_Unmute Output"));
+                        image = gtk_image_new_from_icon_name("audio-volume-medium", GTK_ICON_SIZE_MENU);
```
Comment (zhuyaliang) [inline mate-volume-control/gvc-stream-status-icon.c:238]: It should be used here ```icon->priv->icon_ names[icon->priv->current_ icon]```
Maybe we don't have to judge here
```
g_strdup_printf （"Unmute %s",icon->priv->display_name）
```

File: mate-volume-control/gvc-stream-status-icon.c [hunk 2]
```diff
@@ -234,11 +230,35 @@ on_status_icon_popup_menu (GtkStatusIcon       *status_icon,
...
         gtk_style_context_add_class(context,"mate-panel-menu-bar");
 
-        item = gtk_check_menu_item_new_with_mnemonic (_("_Mute"));
-        gtk_check_menu_item_set_active (GTK_CHECK_MENU_ITEM (item),
-                                        mate_mixer_stream_control_get_mute (icon->priv->control));
+        if (mate_mixer_stream_control_get_mute (icon->priv->control))
+        {
+                if (!g_strcmp0(icon->priv->display_name, _("Output")))
+                {
+                        item = gtk_image_menu_item_new_with_mnemonic(_("_Unmute Output"));
```
Comment (zhuyaliang) [inline mate-volume-control/gvc-stream-status-icon.c:237]: It should be used here ```icon->priv->icon_ names[icon->priv->current_ icon]```
Maybe we don't have to judge here
```
g_strdup_printf （"Unmute %s",icon->priv->display_name）
```

Comment (FreaxMATE) [reply mate-volume-control/gvc-stream-status-icon.c:237]: Sorry I do not really understand what exactly you mean. Could you explain it what could be improved?

**Review by lukefromdc — COMMENTED** [review]
This does function as currently written-and it does have the advantage of making the behavior of the status icon more consistant with that of the newer volume control panel applet.

File: mate-volume-control/gvc-stream-status-icon.c (hunk 2 above)
Comment (zhuyaliang) [reply mate-volume-control/gvc-stream-status-icon.c:237]: There are four states
```
static const gchar *icon_names_output[] = {
        "audio-volume-muted",
        "audio-volume-low",
        "audio-volume-medium",
        "audio-volume-high",
        NULL
};

static const gchar *icon_names_input[] = {
        "audio-input-microphone-muted",
        "audio-input-microphone-low",
        "audio-input-microphone-medium",
        "audio-input-microphone-high",
        NULL
};
```
Like this
```
        if (mate_mixer_stream_control_get_mute (icon->priv->control))
        {
                label = g_strdup_printf ("Unmute %s", icon->priv->display_name);
        }
        else
        {
                label = g_strdup_printf ("Mute %s", icon->priv->display_name);
        }
        item = gtk_image_menu_item_new_with_mnemonic(label);
        image = gtk_image_new_from_icon_name(icon->priv->icon_names[icon->priv->current_icon], GTK_ICON_SIZE_MENU);
```
But I didn't test it，I'll tell you later whether the change was successful

Comment (zhuyaliang) [reply mate-volume-control/gvc-stream-status-icon.c:237]: After testing, the effect is good

Comment (FreaxMATE) [reply mate-volume-control/gvc-stream-status-icon.c]: Thanks man!

**Review by zhuyaliang — APPROVED** [review]

**Review by raveit65 — CHANGES_REQUESTED** [review]
Why not testing building?
```
make[3]: *** [Makefile:770: mate_volume_control_status_icon-gvc-stream-status-icon.o] Error 1
make[3]: Leaving directory '/media/Data/Programme/Linux/Mate-Desktop/git-version/github-matedesktop/mate-media-dist/mate-volume-control'
make[2]: *** [Makefile:519: all] Error 2
make[2]: Leaving directory '/media/Data/Programme/Linux/Mate-Desktop/git-version/github-matedesktop/mate-media-dist/mate-volume-control'
make[1]: *** [Makefile:467: all-recursive] Error 1
make[1]: Leaving directory '/media/Data/Programme/Linux/Mate-Desktop/git-version/github-matedesktop/mate-media-dist'
make: *** [Makefile:399: all] Error 2
```

**Comment by lukefromdc** [comment]
What went different? My build finished fine.We need the linea above where the cut and pasted error message text starts to see what failed.

On 8/1/2021 at 9:55 AM, "raveit65" ***@***.***> wrote:
>
***@***.*** requested changes on this pull request.

**Comment by raveit65** [comment]
Full log with errors:
```
make[3]: Entering directory '/media/Data/Programme/Linux/Mate-Desktop/git-version/github-matedesktop/mate-media-dist/mate-volume-control'
  CC       mate_volume_control_status_icon-gvc-stream-status-icon.o
gvc-stream-status-icon.c: In function 'popup_dock':
gvc-stream-status-icon.c:74:9: warning: 'gtk_status_icon_get_screen' is deprecated [-Wdeprecated-declarations]
   74 |         screen = gtk_status_icon_get_screen (GTK_STATUS_ICON (icon));
      |         ^~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:275,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkstatusicon.h:131:23: note: declared here
  131 | GdkScreen            *gtk_status_icon_get_screen         (GtkStatusIcon      *status_icon);
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c:76:9: warning: 'gtk_status_icon_get_geometry' is deprecated [-Wdeprecated-declarations]
   76 |         if (gtk_status_icon_get_geometry (GTK_STATUS_ICON (icon),
      |         ^~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:275,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkstatusicon.h:166:23: note: declared here
  166 | gboolean              gtk_status_icon_get_geometry       (GtkStatusIcon      *status_icon,
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c: In function 'on_status_icon_popup_menu':
gvc-stream-status-icon.c:235:17: error: 'label' undeclared (first use in this function)
  235 |                 label = g_strdup_printf ("Unmute %s", icon->priv->display_name);
      |                 ^~~~~
gvc-stream-status-icon.c:235:17: note: each undeclared identifier is reported only once for each function it appears in
gvc-stream-status-icon.c:241:9: warning: 'gtk_image_menu_item_new_with_mnemonic' is deprecated: Use 'gtk_menu_item_new_with_mnemonic' instead [-Wdeprecated-declarations]
  241 |         item = gtk_image_menu_item_new_with_mnemonic(label);
      |         ^~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:269,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkimagemenuitem.h:82:12: note: declared here
   82 | GtkWidget* gtk_image_menu_item_new_with_mnemonic (const gchar      *label);
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c:244:9: warning: 'gtk_image_menu_item_set_image' is deprecated [-Wdeprecated-declarations]
  244 |         gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (item), image);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:269,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkimagemenuitem.h:92:12: note: declared here
   92 | void       gtk_image_menu_item_set_image         (GtkImageMenuItem *image_menu_item,
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c:244:9: warning: 'gtk_image_menu_item_get_type' is deprecated: Use 'gtk_menu_item_get_type' instead [-Wdeprecated-declarations]
  244 |         gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (item), image);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:269,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkimagemenuitem.h:76:12: note: declared here
   76 | GType      gtk_image_menu_item_get_type          (void) G_GNUC_CONST;
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c:252:9: warning: 'gtk_image_menu_item_new_with_mnemonic' is deprecated: Use 'gtk_menu_item_new_with_mnemonic' instead [-Wdeprecated-declarations]
  252 |         item = gtk_image_menu_item_new_with_mnemonic (_("_Sound Preferences"));
      |         ^~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:269,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkimagemenuitem.h:82:12: note: declared here
   82 | GtkWidget* gtk_image_menu_item_new_with_mnemonic (const gchar      *label);
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c:255:9: warning: 'gtk_image_menu_item_set_image' is deprecated [-Wdeprecated-declarations]
  255 |         gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (item), image);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:269,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkimagemenuitem.h:92:12: note: declared here
   92 | void       gtk_image_menu_item_set_image         (GtkImageMenuItem *image_menu_item,
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c:255:9: warning: 'gtk_image_menu_item_get_type' is deprecated: Use 'gtk_menu_item_get_type' instead [-Wdeprecated-declarations]
  255 |         gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (item), image);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:269,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkimagemenuitem.h:76:12: note: declared here
   76 | GType      gtk_image_menu_item_get_type          (void) G_GNUC_CONST;
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c:265:9: warning: 'gtk_menu_popup' is deprecated: Use '(gtk_menu_popup_at_widget, gtk_menu_popup_at_pointer, gtk_menu_popup_at_rect)' instead [-Wdeprecated-declarations]
  265 |         gtk_menu_popup (GTK_MENU (menu),
      |         ^~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtklabel.h:34,
                 from /usr/include/gtk-3.0/gtk/gtkaccellabel.h:35,
                 from /usr/include/gtk-3.0/gtk/gtk.h:33,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/gtkmenu.h:138:12: note: declared here
  138 | void       gtk_menu_popup                 (GtkMenu             *menu,
      |            ^~~~~~~~~~~~~~
gvc-stream-status-icon.c:268:25: warning: 'gtk_status_icon_position_menu' is deprecated [-Wdeprecated-declarations]
  268 |                         gtk_status_icon_position_menu,
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:275,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkstatusicon.h:160:23: note: declared here
  160 | void                  gtk_status_icon_position_menu      (GtkMenu            *menu,
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c: In function 'update_icon':
gvc-stream-status-icon.c:391:17: warning: 'gtk_status_icon_set_has_tooltip' is deprecated [-Wdeprecated-declarations]
  391 |                 gtk_status_icon_set_has_tooltip (GTK_STATUS_ICON (icon), FALSE);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:275,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkstatusicon.h:134:23: note: declared here
  134 | void                  gtk_status_icon_set_has_tooltip    (GtkStatusIcon      *status_icon,
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c:394:17: warning: 'gtk_status_icon_set_has_tooltip' is deprecated [-Wdeprecated-declarations]
  394 |                 gtk_status_icon_set_has_tooltip (GTK_STATUS_ICON (icon), TRUE);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:275,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkstatusicon.h:134:23: note: declared here
  134 | void                  gtk_status_icon_set_has_tooltip    (GtkStatusIcon      *status_icon,
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c:417:17: warning: 'gtk_status_icon_set_from_icon_name' is deprecated [-Wdeprecated-declarations]
  417 |                 gtk_status_icon_set_from_icon_name (GTK_STATUS_ICON (icon),
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:275,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkstatusicon.h:106:23: note: declared here
  106 | void                  gtk_status_icon_set_from_icon_name (GtkStatusIcon      *status_icon,
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c:459:9: warning: 'gtk_status_icon_set_tooltip_markup' is deprecated [-Wdeprecated-declarations]
  459 |         gtk_status_icon_set_tooltip_markup (GTK_STATUS_ICON (icon), markup);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:275,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkstatusicon.h:140:23: note: declared here
  140 | void                  gtk_status_icon_set_tooltip_markup (GtkStatusIcon      *status_icon,
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c: In function 'gvc_stream_status_icon_set_icon_names':
gvc-stream-status-icon.c:482:9: warning: 'gtk_status_icon_set_from_icon_name' is deprecated [-Wdeprecated-declarations]
  482 |         gtk_status_icon_set_from_icon_name (GTK_STATUS_ICON (icon), names[0]);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:275,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkstatusicon.h:106:23: note: declared here
  106 | void                  gtk_status_icon_set_from_icon_name (GtkStatusIcon      *status_icon,
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c: In function 'on_status_icon_visible_notify':
gvc-stream-status-icon.c:668:9: warning: 'gtk_status_icon_get_visible' is deprecated [-Wdeprecated-declarations]
  668 |         if (gtk_status_icon_get_visible (GTK_STATUS_ICON (icon)) == FALSE)
      |         ^~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:275,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkstatusicon.h:154:23: note: declared here
  154 | gboolean              gtk_status_icon_get_visible        (GtkStatusIcon      *status_icon);
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~
gvc-stream-status-icon.c: In function 'on_icon_theme_change':
gvc-stream-status-icon.c:677:9: warning: 'gtk_status_icon_set_from_icon_name' is deprecated [-Wdeprecated-declarations]
  677 |         gtk_status_icon_set_from_icon_name (GTK_STATUS_ICON (icon),
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:275,
                 from gvc-stream-status-icon.c:25:
/usr/include/gtk-3.0/gtk/deprecated/gtkstatusicon.h:106:23: note: declared here
  106 | void                  gtk_status_icon_set_from_icon_name (GtkStatusIcon      *status_icon,
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[3]: *** [Makefile:770: mate_volume_control_status_icon-gvc-stream-status-icon.o] Error 1
make[3]: Leaving directory '/media/Data/Programme/Linux/Mate-Desktop/git-version/github-matedesktop/mate-media-dist/mate-volume-control'
make[2]: *** [Makefile:519: all] Error 2
```

**Comment by FreaxMATE** [comment]
Sorry I was sure I built and tested it and it worked but maybe I messed something up. There was the declaration of a `GtkWidget *label` missing but even with this it did not work as expected. The line
`image = gtk_image_new_from_icon_name(icon->priv->icon_names[icon->priv->current_icon], GTK_ICON_SIZE_MENU);`
does not work for this situation because you always want to display the opposite icon in the menu in relation to the tray icon, e.g if audio is not muted, the menu item should show "Mute Output" with the mute icon.
I corrected this and now it should work as expected.

Here is the diff of the previous state of this PR that failed building:
```diff

diff --git a/mate-volume-control/gvc-stream-status-icon.c b/mate-volume-control/gvc-stream-status-icon.c
index 0153365..aeaf268 100644
--- a/mate-volume-control/gvc-stream-status-icon.c
+++ b/mate-volume-control/gvc-stream-status-icon.c
@@ -175,11 +175,7 @@ on_status_icon_button_press (GtkStatusIcon       *status_icon,
 static void
 on_menu_mute_toggled (GtkMenuItem *item, GvcStreamStatusIcon *icon)
 {
-        gboolean is_muted;
-
-        is_muted = gtk_check_menu_item_get_active (GTK_CHECK_MENU_ITEM (item));
-
-        mate_mixer_stream_control_set_mute (icon->priv->control, is_muted);
+        mate_mixer_stream_control_set_mute (icon->priv->control, !mate_mixer_stream_control_get_mute (icon->priv->control));
 }
 
 static void
@@ -234,11 +230,20 @@ on_status_icon_popup_menu (GtkStatusIcon       *status_icon,
         gtk_style_context_add_class(context,"gnome-panel-menu-bar");
         gtk_style_context_add_class(context,"mate-panel-menu-bar");
 
-        item = gtk_check_menu_item_new_with_mnemonic (_("_Mute"));
-        gtk_check_menu_item_set_active (GTK_CHECK_MENU_ITEM (item),
-                                        mate_mixer_stream_control_get_mute (icon->priv->control));
+        if (mate_mixer_stream_control_get_mute (icon->priv->control))
+        {
+                label = g_strdup_printf ("Unmute %s", icon->priv->display_name);
+        }
+        else
+        {
+                label = g_strdup_printf ("Mute %s", icon->priv->display_name);
+        }
+        item = gtk_image_menu_item_new_with_mnemonic(label);
+        image = gtk_image_new_from_icon_name(icon->priv->icon_names[icon->priv->current_icon], GTK_ICON_SIZE_MENU);
+
+        gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (item), image);
         g_signal_connect (G_OBJECT (item),
-                          "toggled",
+                          "activate",
                           G_CALLBACK (on_menu_mute_toggled),
                           icon);
 
```

File: mate-volume-control/gvc-stream-status-icon.c [hunk 3]
```diff
@@ -219,10 +215,11 @@ on_status_icon_popup_menu (GtkStatusIcon       *status_icon,
         GtkWidget *menu;
         GtkWidget *item;
         GtkWidget *image;
+        GtkWidget *label;
```
Comment (zhuyaliang) [inline mate-volume-control/gvc-stream-status-icon.c:218]: It should be ```g_autofree char *label = NULL```;

Comment (FreaxMATE) [reply mate-volume-control/gvc-stream-status-icon.c]: Thanks :)

**Review by raveit65 — APPROVED** [review]
LGTM,
does what it promise

**Pull request merged**